spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -1086,155 +1086,67 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde with CometExprShim
           None
 
         case EqualTo(left, right) =>
-          val leftExpr = exprToProtoInternal(left, inputs)
-          val rightExpr = exprToProtoInternal(right, inputs)
-
-          if (leftExpr.isDefined && rightExpr.isDefined) {
-            val builder = ExprOuterClass.Equal.newBuilder()
-            builder.setLeft(leftExpr.get)
-            builder.setRight(rightExpr.get)
-
-            Some(
-              ExprOuterClass.Expr
-                .newBuilder()
-                .setEq(builder)
-                .build())
-          } else {
-            withInfo(expr, left, right)
-            None
+          createBinaryExpr(left, right, inputs).map { builder =>
+            ExprOuterClass.Expr
+              .newBuilder()
+              .setEq(builder)
+              .build()
```

Comment:
I feel these lines can be a part of `createBinaryExpr` so that we do not have to `map()`?

Reply:
This block of code is specifically for creating `BinaryExpr eq = 9` (note the call to `setEq` that uses the result of `createBinaryExpr`). Each of these 20 similar blocks of code is for setting a specific top-level `expr_struct`. It is only the contained `BinaryExpr` struct that is common to each one. Does that make sense?

Reply:
Hmm .. perhaps I could refactor farther so that the `map` receives an `ExprOuterClass.Expr.Builder` and then only has to call `setEq` in the map block. That would be more concise. I will try that as a follow on PR after https://github.com/apache/datafusion-comet/pull/1056 is merged